app/routes.py

```diff
@@ -1,2 +1,208 @@
+from app import db
 from flask import Blueprint
+from .models.task import Task
+from .models.goal import Goal
+from flask import request
+from flask import jsonify, make_response
+from sqlalchemy import asc, desc 
+import requests
+import os
 
+# creating instance of the model, first arg is name of app's module
+task_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+goal_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+#create a task with null completed at
+@task_bp.route("", methods = ["POST"], strict_slashes = False)
+def create_task():
+    try:
+        request_body = request.get_json()
+        new_task = Task.from_json_to_task(request_body)
+        db.session.add(new_task) # "adds model to the db"
+        db.session.commit() # commits the action above
+        return new_task.to_json_response(), 201
+    except KeyError:
+        return {"details": "Invalid data"}, 400
+
+# Retrieve all /tasks  asc, desc, or id asc
+@task_bp.route("", methods = ["GET"], strict_slashes = False)
+def retrieve_tasks_data():
+    tasks = Task.query.all() 
+    sort_by = request.args.get("sort") # query parameters 
+    tasks_response = []
+
+    if tasks != None: 
+        if sort_by == "asc":  
+            # this is a list (queried by title) in asc order
+            tasks = Task.query.order_by(Task.title.asc()).all() 
+
+        elif sort_by == "desc":
+            # this is a list (queried by title) in desc order
+            tasks = Task.query.order_by(Task.title.desc()).all() 
+        
+        elif sort_by == "id":
+            # list, queried by id in asc order:
+            tasks = Task.query.order_by(Task.id.asc()).all()  
+        
+        tasks_response = [task.task_to_json_response() \
+                for task in tasks]
+    return jsonify(tasks_response), 200
+
+# Retrieve one /task/1     
+@task_bp.route("/<task_id>", methods=["GET"], strict_slashes = False)
+def retrieve_single_task(task_id):
+    task = Task.query.get(task_id)
+    if task != None:
+        # return task.to_json_response(), 200
+        response = task.to_json_response()
+        if task.goal_id:
+            response['task']['goal_id'] = task.goal_id
+        return response, 200
+    return make_response('', 404)
```

Comment:
Consider flipping this so you deal with the edge case first. This pattern can increase readability. I've provided an example of what I mean, but this feedback also applies to the functions below.

```suggestion
    if not Task:
        return make_response('', 404)
    # return task.to_json_response(), 200
    response = task.to_json_response()
    if task.goal_id:
        response['task']['goal_id'] = task.goal_id
    return response, 200
```